plugin-server/src/utils/db/db.ts

```diff
@@ -1453,6 +1453,29 @@ export class DB {
         return selectResult.rows[0]
     }
 
+    public async fetchTeamFromToken(token: string): Promise<Team> {
+        const selectResult = await this.postgresQuery<Team>(
+            `
+            SELECT
+                id,
+                uuid,
+                organization_id,
+                name,
+                anonymize_ips,
+                api_token,
+                slack_incoming_webhook,
+                session_recording_opt_in,
+                ingested_event
+            FROM posthog_team
+            WHERE api_token = $1
```

Comment:
Suggestion: Make sure there's an index for this!

Reply:
checked, we do have it!